projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.ts

```diff
@@ -301,13 +302,23 @@ export class IgxAdvancedFilteringDialogComponent implements AfterViewInit, OnDes
         if (this.editedExpression) {
             this.editedExpression.expression.fieldName = this.selectedColumn.field;
             this.editedExpression.expression.condition = this.selectedColumn.filters.condition(this.selectedCondition);
-            this.editedExpression.expression.searchVal = this.searchValue;
+            this.editedExpression.expression.searchVal = this.transformValue(this.searchValue);
 
             this.editedExpression.inEditMode = false;
             this.editedExpression = null;
         }
     }
 
+    private transformValue(value): any {
```

Comment:
Since this method is copied on 3 different places, we should extract it to the filtering service.